### PR TITLE
GFnet by Jiajun Che s4692111

### PR DESCRIPTION
**Pull request by KevinCheJiajun**
Final version of the asignment

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
**<This is an initial inspection, no action is required at this point.>**

**File Organizing:** Well-organized files.

**Problem Solving:**

- The algorithm solves the problem appropriately. 
- Accuracy in testing dataset: 0.65. 

**Model and functions:**

- It correctly uses PyTorch to construct the GFNet models and functions.
- No data augmentation.

**Code design:** Other parts are good but the predict.py, as the line "model, test_loader = train()" cannot work, so the predict.py will fail in running.

**Code comment and docstring:**

- Good code comments 
- Minimal function docstrings

**Difficulty:** Hard.

**Additional Comments:**

- Insufficient commits, just copy-and-paste code.
- Minimal description of the project, GFNet, experiments, and analysis, etc.
- No validation observation in the training phase, and the test results are suspected to indicate overfitting.

**Comment by aniketgupta17** [comment]
Observational Feedback

Pull Request:
Correctly created the Pull request from Topic Recognition Branch .
The pull request includes no description about the file structure .
No previous feedback incorporated . 

File Organizing: Well-organized files.

Commit Log:
No meaningful commit messages for the project .
Commits should  demonstrate a  more logical development flow.


Documentation:
The README file **should** contain in depth  explanation of the algorithm and the problem it solves.
Code comments and docstrings  are included.
Proper GitHub markdown formatting is used, with organized headings, lists, and code blocks.
Can add about Conclusion and  future improvements in the ReadMe.

**Comment by wangzhaomxy** [comment]
Hi Kevin,

Thanks for your feedback.

I tested your code again, and it is now functional. The error I encountered during my previous test was due to the design of `train.py` and `test.py`, which did not conform to a common pattern that aligns with my testing framework. However, the individual tests are working well now, and I have updated my comments accordingly. Rest assured, this will not impact your final marks, as Shakes will assign those later.

Cheers,
Zhao

**Comment by KevinCheJiajun** [comment]
Thank you very much！
________________________________

**Comment by gayanku** [comment]
## Marking
<table><tbody>
  <tr><td colspan='3'><b>Good/OK/Fair Practice (Design/Commenting, TF/Torch Usage)</b></td> </tr>
	<tr><td></td><td>Good design and implementation. </td><td></td></tr>
	<tr><td></td><td>Spacing and comments. </td><td></td></tr>
	<tr><td></td><td>No Header blocks. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Recognition Problem</b></td></tr>
	<tr><td></td><td>OK solution to problem. Acc low. No validation. Some overfitting.</td><td>-3</td></tr>
	<tr><td></td><td>Driver Script present. </td><td></td></tr>
	<tr><td></td><td>File structure present. </td><td></td></tr>
	<tr><td></td><td>Good Usage & Demo & Visualisation & Data usage. </td><td></td></tr>
	<tr><td></td><td>Module present. </td><td></td></tr>
	<tr><td></td><td>Commenting missing. Minimal doc strings.</td><td>-1</td></tr>
	<tr><td></td><td>Data leakage found. No Data Aug.</td><td>-2</td></tr>
	<tr><td></td><td>Difficulty : Hard.  GFNet (Hard Difficulty)</td><td></td></tr>
	
  <tr><td colspan='3'><b>Commit Log</b></td></tr>
	<tr><td></td><td>Some/Adequate Meaningful commit messages. </td><td>-1</td></tr>
	<tr><td></td><td>Some/Adequate Progressive commits. </td><td>-1</td></tr>
	
  <tr><td colspan='3'><b>Documentation</b></td></tr>
	<tr><td></td><td>Readme :Acceptable. </td><td>-2</td></tr>
	<tr><td></td><td>Model/technical explanation :Good. </td><td></td></tr>
	<tr><td></td><td>Description and Comments :Good. </td><td></td></tr>
	<tr><td></td><td>Markdown used and PDF submitted. PDF Checked.</td><td></td></tr>
	
  <tr><td colspan='3'><b>Pull Request</b></td></tr>
	<tr><td></td><td>Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch). </td><td></td></tr>
	<tr><td></td><td>No Feedback required. </td><td></td></tr>
	<tr><td></td><td>Request Description is missing. Give a  high level intro to the approach.</td><td>-2</td></tr>
  <tr><td colspan='2'><b>TOTAL</b></td><td>-13</td></tr>
</tbody></table>

Marked as per the due date and changes after which aren't necessarily allowed to contribute to grade for fairness.
_Subject to approval from Shakes_

**Comment by shakes76** [comment]
Cant merge because of conflicting changes to main repo files (gitignore). Please update for merge, doesn't affect grade.
